File: smash-ranks-android/app/src/main/java/com/garpr/android/preferences/GeneralPreferenceStore.java
package com.garpr.android.preferences;

import android.support.annotation.NonNull;

import com.garpr.android.models.FavoritePlayer;
import com.garpr.android.models.NightMode;
import com.garpr.android.models.Region;

public interface GeneralPreferenceStore extends BasePreferenceStore {

    @NonNull
    Preference<Region> getCurrentRegion();

    @NonNull
    Preference<FavoritePlayer> getIdentity();

    @NonNull
    Preference<Integer> getLastVersion();

    @NonNull
    Preference<NightMode> getNightMode();

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/preferences/GeneralPreferenceStore.kt
package com.garpr.android.preferences

import com.garpr.android.data.models.FavoritePlayer
import com.garpr.android.data.models.NightMode
import com.garpr.android.data.models.Region

interface GeneralPreferenceStore : PreferenceStore {

    val currentRegion: Preference<Region>

    val hajimeteKimasu: Preference<Boolean>

    val identity: Preference<FavoritePlayer>

    val lastVersion: Preference<Int>

    val nightMode: Preference<NightMode>

}
